tests/rados/test_reads_balancer.py

```diff
@@ -0,0 +1,273 @@
+        if cr_pool.get("pool_type", "replicated") == "erasure":
+            method_should_succeed(
+                rados_obj.create_erasure_pool, name=cr_pool["pool_name"], **cr_pool
+            )
+        else:
+            method_should_succeed(rados_obj.create_pool, **cr_pool)
+        method_should_succeed(rados_obj.bench_write, **cr_pool)
+    log.info("Completed creating pools and writing test data into the pools")
+
+    existing_pools = rados_obj.list_pools_on_cluster()
+
+    log.debug(
+        "Checking the scores on each pool on the cluster, if the pool is replicated pool"
+    )
+    for pool_name in existing_pools:
+        pool_details = rados_obj.get_pool_details(pool=pool_name)
+        log.debug(f"Selected pool name : {pool_name}")
+        if not pool_details["erasure_code_profile"]:
+            log.debug(
+                f"Selected pool is a replicated pool. Checking read balancer scores.\n"
+                f" Fetched from pool {pool_details['read_balance']}"
+            )
+            read_scores = pool_details["read_balance"]
+            if read_scores["score_acting"] > pool_details["size"]:
+                log.error(
+                    f"The balance score on the pool is more than the size of the pool."
+                    f"Balance score : {read_scores['score_acting']}, Size of poool : {pool_details['size']}"
+                )
+                log.error(
+                    "This failure will be ignored because of the bug "
+                    ": https://bugzilla.redhat.com/show_bug.cgi?id=2237352 and the failure will be added post fix"
+                )
+
+                # todo: uncomment the below exception once the bug is fixed
+                # raise Exception("Scores out of bound failure")
+
+            log.debug("Checking if the scores are calculated properly")
+            # Balance_score = raw_score / Optimal_score
+            expected_score = (
+                read_scores["raw_score_acting"] / read_scores["optimal_score"]
+            )
+            if round(read_scores["score_acting"], 2) != round(expected_score, 2):
+                log.error(
+                    f"The scores for the pool is not calculated properly. acting score {read_scores['score_acting']}"
+                    f"Expected : {(read_scores['raw_score_acting'] / read_scores['optimal_score'])}"
+                )
+                raise Exception("Scores out of bound failure")
+            log.info("Completed checking the balance scores for the pool. No errors")
+        else:
+            log.info(
+                f"Selected pool {pool_name} is a EC pool, skipping check of balance scores."
+            )
+        log.info("Completed verification of balance scores for all the pools")
+
+    log.debug(
+        "Setting config to allow clients to perform read balancing on the clusters."
+    )
+    config_cmd = "ceph osd set-require-min-compat-client reef"
+    rados_obj.run_ceph_command(cmd=config_cmd)
```

Comment:
Should the `ceph osd set....` command run with a `-f json` suffix? If not, maybe we can switch to `cephadm.shell([config_cmd])` instead of using `run_ceph_command`

Reply:
It does not matter either way. there is no loss or gain by doing that.